tachyon/zk/plonk/lookup/permute_expression_pair.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_LOOKUP_PERMUTE_EXPRESSION_PAIR_H_
+#define TACHYON_ZK_PLONK_LOOKUP_PERMUTE_EXPRESSION_PAIR_H_
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include "absl/container/btree_map.h"
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/zk/base/evals_pair.h"
+#include "tachyon/zk/plonk/error.h"
+
+namespace tachyon::zk {
+
+template <typename F>
+void FillWithBlind(size_t params_size, size_t blinding_factors,
+                   std::vector<F>& values) {
+  values.reserve(values.size() + blinding_factors + 1);
+  for (size_t i = 0; i < blinding_factors + 1; ++i) {
+    values.push_back(F::Random());
+  }
```

Comment:
Could you refactor this using `base::Parallelize()`?

Reply:
When considering random number generation to be deterministic, it should be computed serially.